### PR TITLE
[DOCS] Re-add docs/en for X-Pack

### PR DESCRIPTION
**Pull request by lcawl**
This PR attempts to re-add the "docs/en" in the conf.yaml for the X-Pack Reference doc builds

### CODE REVIEW

File: conf.yaml [hunk 1]
```diff
@@ -291,20 +291,20 @@ contents:
...
             private:    1
             branches:   [ 5.3, 5.2, 5.1, 5.0 ]
             sources:
               -
                 repo:   x-pack
-                path:   docs/public
+                path:   docs/en
               -
                 repo:   x-pack-kibana
-                path:   docs/public-kb
```
Comment (clintongormley) [inline conf.yaml]: this shouldn't change as you've only renamed public -> en in `x-pack`,  not in `x-pack-kibana`

File: conf.yaml [hunk 2]
```diff
@@ -291,20 +291,20 @@ contents:
...
+                path:   docs/en
                 exclude_branches:   [ 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0]
               -
                 repo:   x-pack-elasticsearch
-                path:   docs/public
```
Comment (clintongormley) [inline conf.yaml]: this shouldn't change as you've only renamed public -> en in `x-pack`,  not in `x-pack-elasticsearch`

File: conf.yaml [hunk 3]
```diff
@@ -291,20 +291,20 @@ contents:
...
+                path:   docs/en
                 exclude_branches: [ 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0]
```
Comment (clintongormley) [inline conf.yaml]: `exclude_branches` should be a subset of `branches`, so you can delete all the 2.x branches

File: conf.yaml [hunk 4]
```diff
@@ -291,20 +291,20 @@ contents:
...
             sources:
               -
                 repo:   x-pack
-                path:   docs/public
+                path:   docs/en
               -
                 repo:   x-pack-kibana
-                path:   docs/public-kb
+                path:   docs/en
                 exclude_branches:   [ 5.3, 5.2, 5.1, 5.0, 2.4, 2.3, 2.2, 2.1, 2.0]
```
Comment (clintongormley) [inline conf.yaml]: `exclude_branches` should be a subset of `branches`, so you can delete all the 2.x branches

File: conf.yaml (hunk 1 above)
Comment (lcawl) [inline conf.yaml]: It's not used by 5.3 or earlier.  But in 5.x and master, it *is* used, and it *is* renamed.

File: conf.yaml (hunk 2 above)
Comment (lcawl) [inline conf.yaml]: Ditto comment above.  It's only used in 5.x and master, where it *is* renamed

File: conf.yaml (hunk 3 above)
Comment (lcawl) [inline conf.yaml]: Thanks, I've committed that change.

**Pull request merged**